Added functions

stangier opened the pull request:
I added the function to show a custom label in front of the hard drive monitor:

![asd](https://cloud.githubusercontent.com/assets/7353918/14067205/8f7ccaac-f45f-11e5-863c-0836f1282b3a.png)

This is called by:

``` ruby
@manager.add_block Barr::Blocks::Hdd.new(device: "sda6", name: "/root")`
```

Commit:
Added function to show label of partition in hdd class

Commit:
Added processes block

stangier commented:
I added the class 'processes' which shows the number of active processes.
Called by:

``` ruby
@manager.add_block Barr::Blocks::Processes.new(bcolor: "#FFAAAA'", fcolor:"#333333")
```

OkayDave commented on lib/barr/blocks/hdd.rb:
```diff
@@ -4,12 +4,13 @@ class Hdd < Block
       def update
         total, used, perc = sys_cmd.chomp.split(" ")
-        @output = "#{used} / #{total} (#{perc})"                                                              
+        @output = "#{@name}: #{used} / #{total} (#{perc})"
```
Nice idea, though the functionality for this is already provided via the `icon` option.

`Barr::Blocks::Hdd.new device: "sda2", icon: "/home: "`

OkayDave commented on lib/barr/blocks/processes.rb:
```diff
@@ -0,0 +1,15 @@
+      end
+
+      def sys_cmd
+        `ps -e | wc -l`
```
Could you call `chomp` after the system cmd please? It's a minor detail as the manager will also remove any newlines as a safety measure, but I'd rather the blocks sanitised their own outputs as much as possible.

OkayDave commented:
Thanks for the PR @derlorenz! 

I've added a couple of in-line comments. I think your `Processes` block would make a great addition. Would you mind writing some specs for it please, and adding documentation for it to the README.

If you're not familiar with specs, have a look at the `/spec/blocks/` folder for examples.

Commit:
Added Battery Class

Commit:
Fixed Process class not updating

stangier commented:
I just added the battery class which shows the batter percentage for laptops. You can pass a true or false argument to set if the remaining time should be shown or not. Also I fixed the process class not updating because it did not have an update function.

I will write specs for it and also an entry to the README.

Commit:
Updated Readme

Commit:
Added processes spec

Commit:
removed binary file from vim

Commit:
Updated .gitignore to not include .swp files from vim

Commit:
Fixed Methos names in battery block

Commit:
added battery spec

OkayDave commented:
@derlorenz thanks for the updates. I've made a few changes to your code to make it more in-line with the projects coding style, and to update it to the newer format. I'm closing your PR and I've opened a new one here: https://github.com/OkayDave/barr/pull/12/files.

Before I can merge it into the main branch, I need to update the Battery spec with a mocked output. Mocking and Stubbing API and system calls is really important as it ensures the specs don't fail due to an API or command being unavailable. In my case, I don't use a laptop so I'm unable to run the specs for the `acpi` call.

Could you post the output of `acpi | cut -d ',' -f 2-3` and `acpi | cut -d ',' -f 2` from your terminal please? I'll then add it to the specs and get your work merged in.

stangier commented:
``` bash
acpi | cut -d "," -f 2-3
```

 96%, 03:17:59 remaining

``` bash
acpi | cut -d "," -f 2
```

 96%
